expression/builtin_time.go

```diff
@@ -3544,6 +3619,33 @@ func (b *builtinAddDateStringRealSig) evalTime(row chunk.Row) (types.Time, bool,
 	return result, isNull || err != nil, err
 }
 
+func (b *builtinAddDateStringRealSig) evalString(row chunk.Row) (string, bool, error) {
+
+	unit, isNull, err := b.args[2].EvalString(b.ctx, row)
```

Comment:
The implementation of `AddDate/SubDate` is almost the same, can we find a way to reuse the code?

Reply:
done